File: src/test/java/no/nav/syfo/consumer/ws/BehandleSakConsumerTest.java
package no.nav.syfo.consumer.ws;

import no.nav.syfo.util.Metrikk;
import no.nav.tjeneste.virksomhet.behandlesak.v1.BehandleSakV1;
import no.nav.tjeneste.virksomhet.behandlesak.v1.informasjon.WSPerson;
import no.nav.tjeneste.virksomhet.behandlesak.v1.informasjon.WSSak;
import no.nav.tjeneste.virksomhet.behandlesak.v1.meldinger.WSOpprettSakRequest;
import no.nav.tjeneste.virksomhet.behandlesak.v1.meldinger.WSOpprettSakResponse;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.ArgumentCaptor;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import org.springframework.test.context.TestPropertySource;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
@TestPropertySource(locations = "classpath:application-test.properties")
public class BehandleSakConsumerTest {
    @Mock
    private BehandleSakV1 behandleSakV1;

    @Mock
    private Metrikk metrikk;

    @InjectMocks
    private BehandleSakConsumer behandleSakConsumer;

    @Test
    public void opprettSak() throws Exception {
        when(behandleSakV1.opprettSak(any())).thenReturn(new WSOpprettSakResponse().withSakId("1"));
        ArgumentCaptor<WSOpprettSakRequest> captor = ArgumentCaptor.forClass(WSOpprettSakRequest.class);

        String sakId = behandleSakConsumer.opprettSak("12345678910");

        verify(behandleSakV1).opprettSak(captor.capture());
        WSSak sak = captor.getValue().getSak();

        assertThat(sakId).isEqualTo("1");
        assertThat(sak.getFagomraade().getValue()).isEqualTo("SYK");
        assertThat(sak.getFagsystem().getValue()).isEqualTo("FS22");
        assertThat(sak.getSakstype().getValue()).isEqualTo("GEN");
        assertThat(sak.getGjelderBrukerListe()).contains(new WSPerson().withIdent("12345678910"));
    }
}

File: src/test/kotlin/no/nav/syfo/consumer/ws/BehandleSakConsumerTest.kt
package no.nav.syfo.consumer.ws

import no.nav.syfo.util.Metrikk
import no.nav.tjeneste.virksomhet.behandlesak.v2.WSAktor
import no.nav.tjeneste.virksomhet.behandlesak.v2.BehandleSakV2
import no.nav.tjeneste.virksomhet.behandlesak.v2.WSOpprettSakRequest
import no.nav.tjeneste.virksomhet.behandlesak.v2.WSOpprettSakResponse
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.ArgumentCaptor
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner
import org.springframework.test.context.TestPropertySource

import org.assertj.core.api.Assertions.assertThat
import org.mockito.ArgumentMatchers.any
import org.mockito.Mockito.verify
import org.mockito.Mockito.`when`

@RunWith(MockitoJUnitRunner::class)
@TestPropertySource(locations = ["classpath:application-test.properties"])
class BehandleSakConsumerTest {
    @Mock
    private val behandleSakV1: BehandleSakV2? = null

    @Mock
    private val metrikk: Metrikk? = null

    @InjectMocks
    private val behandleSakConsumer: BehandleSakConsumer? = null

    @Test
    @Throws(Exception::class)
    fun opprettSak() {
        `when`(behandleSakV1!!.opprettSak(any())).thenReturn(WSOpprettSakResponse().withSakId("1"))
        val captor = ArgumentCaptor.forClass(WSOpprettSakRequest::class.java)

        val sakId = behandleSakConsumer!!.opprettSak("12345678910")

        verify(behandleSakV1).opprettSak(captor.capture())
        val sak = captor.value.sak

        assertThat(sakId).isEqualTo("1")
        assertThat(sak.fagomrade).isEqualTo("SYK")
        assertThat(sak.fagsystem).isEqualTo("FS22")
        assertThat(sak.saktype).isEqualTo("GEN")
        assertThat<WSAktor>(sak.gjelderBrukerListe).contains(WSAktor().withIdent("12345678910"))
    }
}
